src/UI/Components/LogViewer/LogViewer.tsx

```diff
@@ -29,15 +29,24 @@ export interface LogViewerData {
 
 interface LogViewerProps {
   logs: LogViewerData[];
+  defaultSelected?: string;
 }
 
 /**
  * LogViewer component displays logs with a dropdown to select log parts and a download button.
  * Uses PatternFly's LogViewer for rendering logs.
- * @param logs Array of log objects, each with data, id, name, duration, and failed status.
+ *
+ * @prop {LogViewerData[]} logs - Array of log objects, each with data, id, name, duration, and failed status.
+ * @prop {string} defaultSelected - The id of the log to select by default.
+ *
+ * @note If the defaultSelected log is not found in the logs array, the last log will be selected.
+ *
+ * @returns {React.FC<LogViewerProps>} LogViewer component
  */
-export const LogViewerComponent: React.FC<LogViewerProps> = ({ logs }) => {
-  const [selectedLogId, setSelectedLogId] = useState(logs[0]?.id || "");
+export const LogViewerComponent: React.FC<LogViewerProps> = ({ logs, defaultSelected }) => {
+  const [selectedLogId, setSelectedLogId] = useState(
+    defaultSelected || logs[logs.length - 1]?.id || ""
```

Comment:
Hmmm, in views that have lists we often have separate view when the array is empty, `<EmptyView />` I think. Maybe that's something that could be thought about. not strong requirement per se, but I think it would be cleaner

Reply:
you commented the same last time. We are dealing with an array, so we need the safe operator here to target a certain index. The check whether there are logs or not, is already there. We don't show the logViewer if there are no log reports.